src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java

```diff
@@ -0,0 +1,270 @@
+        logger.debug("Getting the resources within " + pkgId + " + package");
+        
+        try {
+            List<String> resIds = new ArrayList<String>();
+            String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/package_show?id=" + pkgId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            JSONArray resources = (JSONArray) result.get("resources");
+            
+            for (Object res : resources) {
+                JSONObject resource = (JSONObject) res;
+                resIds.add((String) resource.get("id"));
+            } // for
+            
+            return resIds;
+        } catch (Exception e) {
+            return null;
+        } // try catch
+    } // getResources
+    
+    /**
+     * Gets the total number of records within a given resource.
+     * @param resId Resource identifier
+     * @return The total number of records within the given resource
+     */
+    public int getNumRecords(String resId) {
+        logger.debug("Getting the number of records within " + resId);
+        
+        try {
+            int numRecords = 0;
+            int i = 0;
+            
+            while (true) {
+                String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                        + "/api/3/action/datastore_search?limit=1000&offset=" + (i * 1000) + "&resource_id=" + resId;
+                CKANResponse resp = doCKANRequest("GET", url, "");
+                JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+                JSONArray records = (JSONArray) result.get("records");
+                
+                if (records.size() == 0) {
+                    break;
+                } // if
+                
+                numRecords += records.size();
+                i++;
+            } // while
+            
+            return numRecords;
+        } catch (Exception e) {
+            return 0;
+        } // try catch
+    } // resURL
+    
+    /**
+     * Gets the records within a given resource.
+     * @param resId
+     * @param start
+     * @param end
+     * @return The records within the given resource
+     */
+    public JSONArray getRecords(String resId, long start, long end) {
+        logger.debug("Getting the [" + start + ", " + end + "] records within " + resId);
+        
+        try {
+            String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/datastore_search?limit=" + end + "&offset=" + start + "&resource_id=" + resId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            return (JSONArray) result.get("records");
+        } catch (Exception e) {
+            return null;
```

Comment:
In case of an exception when executing getRecords(), it seems to me that we may end up getting a RuntimeException since 'records' may be null (see https://github.com/telefonicaid/fiware-tidoop/blob/feature/1_ckan_extensions/src/main/java/com/telefonica/iot/tidoop/hadoop/ckan/CKANRecordReader.java#L71) and no checking about this fact is done at https://github.com/telefonicaid/fiware-tidoop/blob/feature/1_ckan_extensions/src/main/java/com/telefonica/iot/tidoop/hadoop/ckan/CKANRecordReader.java#L78 :)